pkg/kubectl/scale.go

```diff
@@ -81,6 +82,17 @@ func (precondition *ScalePrecondition) Validate(controller *api.ReplicationContr
 	return nil
 }
 
+// ValidateJob ensures that the preconditions match.  Returns nil if they are valid, an error otherwise
+func (precondition *ScalePrecondition) ValidateJob(job *experimental.Job) error {
+	if precondition.Size != -1 && *job.Spec.Parallelism != precondition.Size {
```

Comment:
Parallelism will never be nil, will always be set by [defaults](https://github.com/kubernetes/kubernetes/pull/13259/files#diff-3502a0889593893dcea4fc9488cfff56).